networkx/algorithms/community/modularity_max.py

```diff
@@ -326,6 +326,8 @@ def greedy_modularity_communities(
             raise ValueError(f"best_n must be between 1 and {len(G)}. Got {best_n}.")
         if best_n < cutoff:
             raise ValueError(f"Must have best_n >= cutoff. Got {best_n} < {cutoff}")
+        if best_n == 1:
+            return [set(G)]
```

Comment:
This is maybe not the right place but while reviewing this I realized that currently `greedy_modularity_communities` fails with a cryptic `ValueError: cutoff must be between 1 and 0. Got {something between 1 and 0}.` when a graph with no nodes is passed and it pops up with a `ZeroDivisionError` if there are no edges. We should probably catch them and make it more graceful.

I can create a new issue for this :)

Reply:
Yes!  Those corner cases certainly are not covered here.  
And there many be others...  3-nodes and 1 edge?